starter_packs/views.py

```
@@ -671,3 +690,303 @@ def follow_bg(user_id: int, starter_pack_slug: str):
+    media_dir = settings.BASE_DIR / "media"
+    splash_dir = settings.BASE_DIR / "media" / "splash"
+
+    for directory in (media_dir, splash_dir):
+        try:
+            directory.mkdir(mode=0o770, exist_ok=True)
+        except (FileExistsError, PermissionError) as e:
+            # With `exist_ok=True`, the FileExistsError is only raised if the media path
+            # already exists and is _not_ a directory.
+            logger.exception("Unable to create %s directory: %s", directory, type(e).__name__)
+            return
+        try:
+            write_test_path = directory / "__write_test.lock"
+            write_test_path.touch()
+            write_test_path.unlink()
+        except OSError:
+            logger.exception("Unable to write to directory %s", directory)
+            return
+
+    render_resolution = (resolution[0] * supersampling_factor, resolution[1] * supersampling_factor)
+    font_path = settings.BASE_DIR / "static" / "InterVariable.ttf"
+    attribution_font = ImageFont.truetype(font_path, round(30 * supersampling_factor))
+
+    image = get_splash_background(
+        render_resolution[0],
+        render_resolution[1],
+        f"Hosted by {host_attribution}",
+        attribution_font,
+        media_dir / f"starterpack_bg_{render_resolution[0]}x{render_resolution[1]}.png",
+    )
+
+    margin_top = 2.5 * 48 * supersampling_factor
+    margin_bottom = 2 * 36 * supersampling_factor
+    margin_side = 0.05 * resolution[0] * supersampling_factor
+
+    accounts = get_splash_image_accounts(starter_pack)
+    random.shuffle(accounts)
+
+    inner_width = render_resolution[0] - 2 * margin_side
+    inner_height = render_resolution[1] - margin_top - margin_bottom
+    inner_aspect_ratio = inner_width / inner_height
+
+    num_lines = 1 + math.floor(math.sqrt(max(len(accounts) - 1, 1) / inner_aspect_ratio))
+    per_line = math.floor(len(accounts) / num_lines)
+    num_lines = math.floor(len(accounts) / per_line)
```

If a starter pack doesn't have any accounts this line errors out with:

```
  File "/Users/anzepecar/code/fedidevs/starter_packs/management/commands/update_splash_images.py", line 82, in render_splash_images_for
    render_splash_image(starter_pack, host_attribution)
    ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/anzepecar/code/fedidevs/starter_packs/views.py", line 873, in render_splash_image
    num_lines = math.floor(len(accounts) / per_line)
                           ~~~~~~~~~~~~~~^~~~~~~~~~
ZeroDivisionError: division by zero
```

---

Fair, I didn't test that case.

Thinking about it, what *should* we do if a starter pack has no accounts in it? We could render an image with just the title and gradient background, but it would look very empty. Or we could have `render_splash_image` fail fast so that the starter pack will have no splash image and the UI and API can fall back to the generic one.